### PR TITLE
being knp voter fix

### PR DESCRIPTION
**Pull request by ob1201**
| Deprecations? | yes

| Fixed tickets | comma-separated list of tickets fixed by the PR, if any
#303

### CODE REVIEW

File: src/Voter/UriPrefixVoter.php [hunk 1]
```diff
@@ -51,6 +76,11 @@ public function matchItem(ItemInterface $item)
...
         $content = $item->getExtra('content');
 
+        $request = $this->request;
+        if (null !== $this->requestStack) {
+            $request = $this->requestStack->getMasterRequest();
+        }
+/* no idea what to do here ... & beyond */        
         if ($content instanceof Route && $content->hasOption('currentUriPrefix')) {
             $currentUriPrefix = $content->getOption('currentUriPrefix');
             $currentUriPrefix = str_replace('{_locale}', $this->request->getLocale(), $currentUriPrefix);
```
Comment (ElectricMaxxx) [inline src/Voter/UriPrefixVoter.php]: You should replace each ocurence of `$this->request` (like here) by your request.

**Comment by ElectricMaxxx** [comment]
Btw: you can run the tests locally by calling `make test` (after a composer update for sure)

File: src/Voter/UriPrefixVoter.php [hunk 2]
```diff
@@ -51,6 +76,11 @@ public function matchItem(ItemInterface $item)
 
         $content = $item->getExtra('content');
 
+        $request = $this->request;
+        if (null !== $this->requestStack) {
+            $request = $this->requestStack->getMasterRequest();
```
Comment (dbu) [inline src/Voter/UriPrefixVoter.php]: i would do a private getRequest method that looks at stack and falls back to the request from the setter.

**Comment by dbu** [comment]
continued in #305, thanks for reporting and getting this started @ob1201 !

**Pull request closed without merging**